tango/test_context.py

```diff
@@ -192,34 +332,49 @@ def post_init(self):
                 "The post_init routine failed to report anything")
             self.queue.put((None, exc, None))
 
-    def generate_db_file(self, server, instance, device,
-                         tangoclass=None, properties={}):
-        """Generate a database file corresponding to the given arguments."""
-        if not tangoclass:
-            tangoclass = server
+    def append_db_file(self, server, instance, tangoclass, device_prop_info):
+        """Generate a database file corresponding to the given arguments.
+        """
+        device_names = [info["name"] for info in device_prop_info]
         # Open the file
-        with open(self.db, 'w') as f:
+        with open(self.db, "a") as f:
             f.write("/".join((server, instance, "DEVICE", tangoclass)))
-            f.write(': "' + device + '"\n')
+            for device_name in device_names:
+                f.write(': "' + device_name + '"\n')
+            f.flush()
         # Create database
         db = Database(self.db)
-        # Patch the property dict to avoid a PyTango bug
-        patched = dict((key, value if value != '' else ' ')
-                       for key, value in properties.items())
         # Write properties
-        db.put_device_property(device, patched)
+        for info in device_prop_info:
+            device_name = info["name"]
+            properties = info.get("properties", {})
+            # Patch the property dict to avoid a PyTango bug
+            patched = dict((key, value if value != '' else ' ')
+                           for key, value in properties.items())
+            db.put_device_property(device_name, patched)
         return db
 
-    def get_device_access(self):
-        """Return the full device name."""
-        form = 'tango://{0}:{1}/{2}#{3}'
-        return form.format(self.host, self.port, self.device_name, self.nodb)
-
     def get_server_access(self):
         """Return the full server name."""
         form = 'tango://{0}:{1}/{2}#{3}'
         return form.format(self.host, self.port, self.server_name, self.nodb)
 
+    def get_device_access(self, device_name):
+        """Return the full device name."""
+        form = 'tango://{0}:{1}/{2}#{3}'
+        return form.format(self.host, self.port, device_name, self.nodb)
+
+    def get_device(self, device_name):
+        """Return the device proxy corresponding to the given device name.
+
+        Maintains previously accessed device proxies in a cache to not recreate
+        then on every access.
+        """
+        if not device_name in self._devices:
```

Comment:
Minor point, but it will be slightly easier to read as:  `if device_name not in self._devices:`

Reply:
Done in d30daad5.